playwright/pages/components/EnvironmentSelector.ts

```diff
@@ -0,0 +1,61 @@
+import { expect, Locator, Page } from "@playwright/test";
+
+export class EnvironmentSelector {
+  readonly page: Page;
+  readonly loginTextButton: Locator;
+  readonly loginIconButton: Locator;
+  readonly environmentName: Locator;
+  readonly dropdownTrigger: Locator;
+
+  constructor(page: Page) {
+    /**
+     * Components
+     */
+    this.page = page;
+
+    /**
+     * Static locators
+     */
+    this.loginTextButton = page.getByRole("button", {
+      name: "Login required",
+      exact: true,
+    });
+    // TODO: Replace with `getByRole` once `<IconButton>` supports labels.
```

Comment:
⚠️ Maybe you could create a ticket for that and put ths ticket number like:
`// TODO(DT-1234): Replace with getByRole once <IconButton> supports labels.`

Same for others. WDYT?

Reply:
Great idea! I'll update the comments with tickets.